qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py

```diff
@@ -539,6 +692,8 @@ def init_measurement(self) -> None:
 
     def reset(self) -> None:
         self.write('*RST')
+        self.ask('*OPC?')
+        self.snapshot(update=True)
```

Comment:
Could you clarify why this change is needed? What does opc call and snapshot call in this case?

Reply:
We always want the snapshot to stay up to date with the parameters. Calling `reset` potentially changes all values, so we better update our beliefs of their values.

`*OPC?` ensures that the instrument is done resetting before we start querying the parameters for their values. Without that line, we get the old values.

Reply:
sweet. thank you!